lib/poseidon/sync_producer.rb

```diff
@@ -42,7 +42,12 @@ def send_messages(messages)
       messages_to_send = MessagesToSend.new(messages, @cluster_metadata)
 
       if refresh_interval_elapsed?
-        refresh_metadata(messages_to_send.topic_set)
+        begin
+          refresh_metadata(messages_to_send.topic_set)
+        rescue Errors::UnableToFetchMetadata => e
```

Comment:
I haven't traced through all of the code, but my only comment is making sure this is the only error that could possibly bubble up here